scripts/aad/create_application_administrator.sh

```diff
@@ -0,0 +1,191 @@
+#!/bin/bash
+
+# Setup Script
+set -euo pipefail
+# AZURE_CORE_OUTPUT=jsonc # force CLI output to JSON for the script (user can still change default for interactive usage in the dev container)
+
+function show_usage()
+{
+    cat << USAGE
+
+Utility script for creating an application administrator for TRE. This is optional and is normal
+if you want to delegate Application creation to TRE.
+This script is trigger by the environment variable AUTO_WORKSPACE_APP_REGISTRATION.
+You must be logged in using Azure CLI with sufficient privileges to modify Azure Active Directory to run this script.
+
+Usage: $0 [--admin-consent]
+
+Options:
+    -a,--admin-consent          Optional, but recommended. Grants admin consent for the app registrations, when this flag is set.
+                                Requires directory admin privileges to the Azure AD in question.
+
+USAGE
+    exit 1
+}
+
+if ! command -v az &> /dev/null; then
+    echo "This script requires Azure CLI" 1>&2
+    exit 1
+fi
+
+# Get the directory that this script is in
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+
+declare grantAdminConsent=0
+declare currentUserId=""
+declare spId=""
+declare msGraphUri="https://graph.microsoft.com/v1.0"
+declare appName="sp_application_admin"
```

Comment:
Feel this should be TRE specific so know which TRE carried out the operation.

Reply:
Done!